server/routes/events.js

```diff
@@ -12,218 +13,580 @@ router.get('/ping', function(req, res, next) {
+				}
+
+				return trx.commit;
+			})
+	})
+	.then(function() {
+  	res.send(util.message('Successfully updated event_id: ' + event_id));
+  })
+  .catch(err => { return next(err) });
+});
+
+// Delete an event by event_id
+router.delete('/:event_id', function(req, res, next) {
   const event_id = req.params.event_id;
-  db.query('SELECT * FROM event WHERE fb_id = $1', [event_id], (err, result) => {
-		if (err) return next(err);
-  	res.send(result.rows);
-  });
+  knex('event').del().where({ fb_id: event_id })
+    .then(result => {
+      if (result) {
+        res.send(util.message('Successfully deleted event with event_id ' + event_id));
+      } else {
+        util.throwError(404, 'No event found to delete');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET all users that have favorited the given event
 router.get('/:event_id/favorites', function(req, res, next) {
   const event_id = req.params.event_id;
-  db.query('SELECT user_id FROM favorite_events WHERE event_id = $1', [event_id], (err, result) => {
-		if (err) return next(err);
-  	res.send(result.rows);
-  });
+  knex('favorite_events').select('user_id').where({ event_id: event_id })
+    .then(result => {
+      if(result.length) {
+        res.json(result);
+      } else {
+        res.status(404).json('No events where fb_id = ' + event_id + ' found.');
+      }
+    })
+    .catch(err => { return next(err) });
+});
+
+// Update user registration for a given event
+router.put('/:event_id/register/:user_id', function(req, res, next) {
+  const event_id = req.params.event_id;
+  const user_id = req.params.user_id;
+  const paid = req.body.paid;
+  if (paid == "true" || paid == "false") {
+    knex('event_registration').update({ has_paid: paid })
+      .where({ event_id: event_id, user_id: user_id })
+      .then(result => {
+        if (result) {
+          res.send(util.message('Successfully updated user registration for event with event_id ' + event_id));
+        } else {
+          util.throwError(404, 'No user found with user_id ' + user_id + ' for event with event_id ' + event_id);
+        }
+      })
+      .catch(err => { return next(err) });
+  } else {
+    util.throwError(400, 'Parameter paid is not a boolean')
+  }
 });
 
 // GET all users registered and have paid or not paid for a given event
 router.get('/:event_id/register', function(req, res, next) {
   const event_id = req.params.event_id;
   const paid = req.query.paid;
-  if (paid == undefined) {
-		db.query('SELECT user_id FROM event_registration WHERE event_id = $1', [event_id], (err, result) => {
-	  	if (err) return next(err);
-    	res.send(result.rows);
-  	});
-  } else if (paid == "true" || paid == "false") {
-		db.query('SELECT user_id FROM event_registration WHERE event_id = $1 AND has_paid = $2', [event_id, paid], (err, result) => {
-			if (err) return next(err);
-      res.send(result.rows);
-  	});
+
+  if (!paid || paid == "true" || paid == "false") {
```

Comment:
If paid is invalid/not set, we want to default to returning all checked in/registered users. But let's avoid redundant query code.

Also make sure to lowercase paid (idk what the function is, probs paid.lower()???)

store the main part of the query in a variable

let event_registration_query = knex('event_registration').select('user_id')
if (paid == 'true' || paid == 'false') {
  event_registration_query = event_registration_query.where({ event_id: event_id, has_paid: paid })
} else {
  event_registration_query = event_registration_query.where({ event_id: event_id })
}

then have the whole
event_registration_query.then(...)